src/compartment/AgentContainer.java

```diff
@@ -510,7 +510,8 @@ public void addAgent(Agent agent)
 		else
 		{
 			this._agentList.add(agent);
-			agent.simplifyLocation();
+			// Should we do this? Needs review
```

Comment:
Probably we can call this method before/during reintroduction into a spatial compartment. I suggest leaving the method for later use, but indeed we can remove the call here. Good catch